### PR TITLE
improvements and misc stuff

### PR DESCRIPTION
**Pull request by uncomfyhalomacro**

### CODE REVIEW

**Comment by uncomfyhalomacro** [comment]
I subconsciously made outdir as Option<PathBuf>. It won't make sense so I reverted it back. Im dumb.

File: src/utils.rs [hunk 1]
```diff
@@ -55,3 +56,20 @@ fn copy_dir_all(src: impl AsRef<Path>, dst: impl AsRef<Path>) -> std::io::Result
...
+
+pub fn get_compression_type(file: impl AsRef<Path>) -> Result<Compression, ()> {
+    match &file.as_ref().extension() {
+        Some(ext) => {
+            let ext = ext.to_str().expect("Failed to convert to str");
+            match ext {
+                "zst" => Ok(Compression::Zst),
+                "zstd" => Ok(Compression::Zst),
+                "gz" => Ok(Compression::Gz),
+                "zip" => Ok(Compression::Zip),
```
Comment (Firstyear) [inline src/utils.rs:68]: Is it even worth support zip?

**Review by Firstyear — APPROVED** [review]

File: src/utils.rs (hunk 1 above)
Comment (uncomfyhalomacro) [reply src/utils.rs:68]: probably not. we can remove that until someone opens an issue. I will push later some changes

File: src/main.rs [hunk 2]
```diff
@@ -1,8 +1,35 @@
...
+                process_srctar(&srcpath, &compression_type);
+            } else if matches!(kind, SrcKind::SrcDir) {
+                let srcpath = args
+                    .srcdir
+                    .as_deref()
+                    .expect("Source dir cannot be determined");
+                process_srcdir(&srcpath);
+            }
+        }
+        None => panic!("Not satisfied"),
```
Comment (Firstyear) [inline src/main.rs:29]: Perhaps a return here for std::process::ExitCode might be better with a proper error/warning.

As a good point as well, consider using tracing from early on for logging :)

Comment (Firstyear) [reply src/main.rs:29]: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/index.html#composing-layers

**Comment by Firstyear** [comment]
I see no reason not to merge this btw, especially so early in dev :)

File: src/main.rs (hunk 2 above)
Comment (uncomfyhalomacro) [reply src/main.rs:29]: Thanks! I will use tracing :D

**Comment by uncomfyhalomacro** [comment]
Yeah. I will merge this for now. 😄

**Pull request merged**